demo/apisix/tmail-apisix-plugin-runner/src/test/java/com/linagora/apisix/plugin/RevokedTokenRepositoryContract.java

```diff
@@ -0,0 +1,51 @@
+package com.linagora.apisix.plugin;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+interface RevokedTokenRepositoryContract {
+
+    IRevokedTokenRepository testee();
+
+    @Test
+    default void existShouldReturnFalseWhenDoesNotExist() {
+        assertThat(testee().exist("sid1")).isFalse();
+    }
+
+    @Test
+    default void existShouldReturnTrueWhenExist() {
+        String sid = "sid1";
+        testee().add(sid);
+        assertThat(testee().exist(sid)).isTrue();
+    }
+
+    @Test
+    default void existShouldBeIdempotent() {
```

Comment:
such test does not make sense, it's a get after all on something in redis. I would expect more a test about addShouldBeIdempotent ?